app/controllers/hq/sponsorships_controller.rb

```diff
@@ -0,0 +1,23 @@
+class Hq::SponsorshipsController < HqController
+
+  def inactivate
+    sponsorship = Sponsorship.find(params[:id])
+    if sponsorship.inactivate params[:sponsorship][:end_date]
```

Comment:
Sorry, all this needs changing now that service objects PR has been merged in. On the plus side, I tried to make the AA sponsorships controller code as portable as possible, so you should be able to mostly copy-paste it in here.